### DISCUSSION TITLE
Attribute to add tag or config to a specific service only?

### DISCUSSION THREAD

**Discussion by donquixote** [symfony/symfony#57198] [Q&A]
The `#[Autoconfigure(...)]` and `#[AutoconfigureTag(...)]` attributes can apply configuration to all services where the class implements or extends the base class or interface.

Sometimes I only want to apply configuration to a specific service, not to its children.
E.g. I might want to make a specific service public, or add a tag.

So far I manage to do this by "abusing" the attributes mentioned above, and putting them on the class directly.
This is fine if the class has no children. But if it does have children, now they will also get the configuration or tag.

I could create my own attribute and then call `$containerBuilder->registerAttributeForAutoconfiguration()`.
But it seems odd that such an attribute does not already exist in symfony.

**Comment by donquixote** [comment]
Also, I can use `AsAlias(public: true)` to make the interface alias public.
But this is not really the same.

**Comment by latenzio** [comment]
I think https://symfony.com/doc/current/service_container/parent_services.html is what you are looking for. You can define tags on the extending service separately.
